core/src/main/java/org/pentaho/platform/web/http/api/resources/GenericFileResource.java

```diff
@@ -286,18 +287,29 @@ public Response createFile( @NonNull @PathParam( "path" ) String path,
 
       if ( !fileCreated ) {
         // File already exists and overwrite is false - return conflict
-        throw new WebApplicationException( "File already exists. Use overwrite=true to replace it.",
-          Response.Status.CONFLICT );
+        return Response.status( Response.Status.CONFLICT )
+          .entity( Messages.getString( "GenericFileResource.FILE_EXISTS_OVERWRITE" ) )
+          .type( MediaType.APPLICATION_JSON )
+          .build();
       }
 
       return Response.status( Response.Status.CREATED ).build(); // 201 for new file
 
     } catch ( InvalidPathException | InvalidOperationException e ) {
-      throw new WebApplicationException( e, Response.Status.BAD_REQUEST );
+      return Response.status( Response.Status.BAD_REQUEST )
+        .entity( e.getMessage() )
+        .type( MediaType.APPLICATION_JSON )
+        .build();
     } catch ( AccessControlException e ) {
-      throw new WebApplicationException( e, Response.Status.FORBIDDEN );
+      return Response.status( Response.Status.FORBIDDEN )
+        .entity( e.getMessage() )
+        .type( MediaType.APPLICATION_JSON )
+        .build();
     } catch ( OperationFailedException e ) {
-      throw new WebApplicationException( e, Response.Status.INTERNAL_SERVER_ERROR );
+      return Response.status( Response.Status.INTERNAL_SERVER_ERROR )
+        .entity( e.getMessage() )
+        .type( MediaType.APPLICATION_JSON )
+        .build();
```

Comment:
As previously discussed, I think this use of directly outputting exception messages to the client merits at a minimum changing the documentation of the various exception classes to note that the message of the exceptions should not contain any sensitive information.

It would be best to also remove the constructors with `Throwable`, given how these pass that exception's message (uncontrolled) to the message.